test/utils/sesar_api_test.py

```diff
@@ -0,0 +1,87 @@
+
+import inspect
+from pytest import raises
+
+from sample_uploader.utils.sesar_api import (
+    _get_igsn_endpoint,
+    retrieve_sample,
+)
+
+
+def start_test():
+    testname = inspect.stack()[1][3]
+    print('\n*** starting test: ' + testname + ' **')
+
+
+def fail_retrieve_sample(igsn, error_msg, error_type, contains=True):
+
+    with raises(Exception) as context:
+        retrieve_sample(igsn)
+
+    assert type(context.value) == error_type
+
+    if contains:
+        assert error_msg in str(context.value.args[0])
+    else:
+        assert error_msg == str(context.value.args[0])
+
+
+def test__get_igsn_endpoint():
+
+    start_test()
+
+    igsn_endpoint = _get_igsn_endpoint()
+    expected_igsn_endpoint = 'https://app.geosamples.org/sample/igsn'
+
+    assert igsn_endpoint == expected_igsn_endpoint
+
+
+def test_retrieve_sample_fail():
+
+    start_test()
+
+    igsn = 'fake_isgn'
+    fail_retrieve_sample(igsn, '400:Invalid IGSN', RuntimeError)
+
+    igsn = 'IEAWH9999'  # TODO: this ISGN might become a valid ISGN in the future
+    fail_retrieve_sample(igsn, f'404:{igsn} is not registered in SESAR',
+                         RuntimeError)
+
+
+def test_retrieve_sample():
+
+    start_test()
+
+    igsn = 'IEAWH0001'
+    sample = retrieve_sample(igsn)
+
+    expected_sample_metadata = ['qrcode_img_src', 'user_code', 'igsn', 'name', 'sample_type',
+                                'publish_date', 'material', 'collection_method', 'purpose',
+                                'latitude', 'longitude', 'elevation_unit', 'navigation_type',
+                                'primary_location_type', 'primary_location_name',
+                                'location_description', 'locality_description',
+                                'cruise_field_prgrm', 'collector', 'collection_start_date',
+                                'collection_date_precision', 'current_archive',
+                                'current_archive_contact']
+    assert set(sample.keys()) == set(expected_sample_metadata)
+
+    assert sample['igsn'] == igsn
```

Comment:
Nit: consider using dictionary equality to simplify this list to one assertion 

https://stackoverflow.com/questions/53348959/python3-determine-if-two-dictionaries-are-equal

If the retrieved sample has excess keys you need to ignore, you could extract the desired keys like `{key:sample[key] for key in expected_sample_metadata}`

Reply:
👍